### PR TITLE
Write all the things

### PR DESCRIPTION
**Pull request by terw-dan**
Adds an assessment to write about all the things people should include in their text.

### CODE REVIEW

File: js/src/assessments/allTheThingsAssessment.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+				result.score = 6;
+				result.text = "You know the Illuminati are involved. Don't be afraid to mention them as well!!";
+				break;
+			case "freemasons":
+				result.score = 6;
+				result.text = "Do you really think the Illuminati are alone? Of course not, the Freemasons are in on it as well!";
+				break;
+			case "reptilians":
+				result.score = 6;
+				result.text = "Our earth is controlled by Reptillians, you know that. Please don't be too hesitant too tell that!!";
```
[the remark this answers is not in the document]
Comment (terw-dan) [reply js/src/assessments/allTheThingsAssessment.js]: Ok, an extra ! for you ;)

File: js/src/researches/allTheThings.js [hunk 2]
```diff
@@ -4,8 +4,8 @@ let toRegex = require( "yoastseo/js/stringProcessing/createWordRegex.js" );
 /**
  * Finds the missing thing in the text. Checks against the list for things to match.
  *
- * @param text
- * @returns {string} The first
+ * @param {string} text The text to check for mentions in the list
```
Comment (IreneStr) [inline js/src/researches/allTheThings.js]: Please add periods at the end of this and the next description

File: js/src/researches/allTheThings.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+let allTheThings = [ "moon landing", "earth is flat", "ufo", "area 51", "denver airport", "illuminati", "freemasons", "reptilians", "chemtrails"  ];
+let toRegex = require( "yoastseo/js/stringProcessing/createWordRegex.js" );
+
+
+const matchAllThethings = function( text ) {
+	let i, missingMatch = "";
```
[the remark this answers is not in the document]
Comment (IreneStr) [reply js/src/researches/allTheThings.js]: rename to missingThing

File: js/src/researches/allTheThings.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+let allTheThings = [ "moon landing", "earth is flat", "ufo", "area 51", "denver airport", "illuminati", "freemasons", "reptilians", "chemtrails"  ];
```
[the remark this answers is not in the document]
Comment (IreneStr) [reply js/src/researches/allTheThings.js]: ^

File: js/src/tests/allTheThings.test.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+		expect( thing ).toEqual( "moon landing" );
+	});
+
+	it ( "returns flat earth as the next thing to write about", function(){
+		let mockPaper = new Paper( "The moon landing was staged" );
+		let thing = allTheThings( mockPaper );
+		expect( thing ).toEqual( "earth is flat" );
+	});
+
+	it ( "returns ufo as the next thing to write about", function(){
```
[the remark this answers is not in the document]
Comment (IreneStr) [reply js/src/tests/allTheThings.test.js]: ^

File: js/src/tests/allTheThings.test.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+let allTheThings = require( "../researches/allTheThings" );
+let Paper = require( "yoastseo/js/values/Paper.js" );
+
+describe( "This will check if you've written all the things people need to know!", function(){
+	it ( "returns the moonlanding as the next thing to write about", function(){
+		let mockPaper = new Paper( "This is an ordinary text." );
+		let thing = allTheThings( mockPaper );
```
[the remark this answers is not in the document]
Comment (IreneStr) [reply js/src/tests/allTheThings.test.js]: ^

File: js/src/tests/allTheThings.test.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+		expect( thing ).toEqual( "ufo" );
+	});
+
+	it ("returns an empty string when everything is written.", function(){
+		let mockPaper = new Paper( "moon landing earth is flat ufo area 51 denver airport illuminati freemasons reptilians chemtrails" );
+		let thing = allTheThings( mockPaper );
+		expect( thing ).toEqual( "" );
+	});
+});
```
[the remark this answers is not in the document]
Comment (IreneStr) [reply js/src/tests/allTheThings.test.js]: ^

File: js/src/researches/allTheThings.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+let toRegex = require( "yoastseo/js/stringProcessing/createWordRegex.js" );
+
+/**
+ * Finds the missing thing in the text. Checks against the list for things to match.
+ *
+ * @param {string} text The text to check for mentions in the list
```
Comment (IreneStr) [inline js/src/researches/allTheThings.js]: Please add periods at the end of this and the next description

File: js/src/tests/allTheThingsAssessment.test.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+	});
+
+	it ( 'returns a score of 3 when only the moon landing was mentioned', () => {
+		let mockPaper = new Paper( "This is a simple text about: moon landing earth is flat ufo area 51 denver airport." + filler );
+		let result = allTheThingsAssessment.getResult( mockPaper );
+
+		expect( result.getScore() ).toBe( 6 );
+		expect( result.getText() ).toBe( "You know the Illuminati are involved. Don't be afraid to mention them as well!!" );
+	});
+
```
[the remark this answers is not in the document]
Comment (IreneStr) [reply js/src/tests/allTheThingsAssessment.test.js]: Has been added 👍

File: js/src/tests/allTheThingsAssessment.test.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+	it ( 'returns a score of 6 when the last thing mentioned from the list is Denver Airport', () => {
+		let mockPaper = new Paper( "This is a simple text about: moon landing earth is flat ufo area 51 denver airport." + filler );
+		let result = allTheThingsAssessment.getResult( mockPaper );
+
+		expect( result.getScore() ).toBe( 6 );
+		expect( result.getText() ).toBe( "You know the Illuminati are involved. Don't be afraid to mention them as well!!" );
+	});
+
+
+	it ( 'returns a score of 6 when the illuminati aren\'t mentioned, but chemtrails are.', () => {
```
Comment (IreneStr) [inline js/src/tests/allTheThingsAssessment.test.js]: Returns a score of 6 when something in the middle of the list (Denver Airport) and something at the end of the list (chemtrails) are mentioned, but things in between aren't.

**Comment by IreneStr** [comment]
CR and acceptance done 👍

**Pull request merged**